packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1054,6 +1068,62 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Updates, re-authorizes and then restarts given snap.
+   *
+   * @param snapId The id of the Snap to be updated
+   * @param newVersionRange A semver version range in which the maximum version will be chosen
+   * @returns @type {TruncatedSnap} if updated, @type {null} otherwise
+   */
+  async updateSnap(
+    snapId: SnapId,
+    newVersionRange: string = DEFAULT_REQUESTED_SNAP_VERSION,
```

Comment:
Do we need to use `resolveVersion` somewhere here for backwards compatibility with `latest`?

Reply:
Seems `resolveVersion` is only used when getting direct input from JSON-RPC, and `DEFAULT_REQUESTED_SNAP_VERSION` is used otherwise. This function is not directly called from RPC

Reply:
Yeah, that seems reasonable to me.